gemrb/core/EffectQueue.cpp

```diff
@@ -1736,33 +1730,31 @@ const Effect *EffectQueue::HasEffectWithParamPair(EffectRef &effect_reference, i
 }
 
 //this could be used for stoneskins and mirror images as well
-void EffectQueue::DecreaseParam1OfEffect(ieDword opcode, ieDword amount)
+bool EffectQueue::DecreaseParam1OfEffect(ieDword opcode, ieDword amount)
 {
+	bool found = false;
 	for (auto& fx : effects) {
 		MATCH_OPCODE()
 		MATCH_LIVE_FX()
-		ieDword value = fx.Parameter1;
-		if( value>amount) {
-			value -= amount;
-			amount = 0;
-		} else {
-			amount -= value;
-			value = 0;
-		}
-		fx.Parameter1 = value;
-		if (value) {
-			return;
```

Comment:
The removal of this short-circuit affects other users as well, but I don't think it makes a difference. They're chaos shield, both stoneskins and mirror image, none of which one can stack AFAIK.

I suggest you add a comment, so later down the line nobody "optimizes" the loop with a break when `found` gets set to true — it intentionally affects all effects of the desired opcode. You can just delete the existing one at the top of the function and do it there.

Reply:
Can do!